packages/node/src/lib/ReadMe.ts

```diff
@@ -0,0 +1,212 @@
+          console.error(e.data);
+        }
+        // Don't want to cause an error in their API
+        return next();
+      }
+    }
+
+    if (req.path === '/readme-webhook' && req.method === 'POST' && !options.disableWebhook) {
+      try {
+        verifyWebhook(req.body, req.headers['readme-signature'] as string, readmeProjectData.jwtSecret as string);
+        const user = await userFunction(req, getUser);
+        return res.send(user);
+      } catch (e) {
+        return res.status(400).json({ error: (e as Error).message });
+      }
+    } else if (req.path === '/readme-setup' && options.development) {
+      const setupHtml = buildSetupView({
+        baseUrl,
+        subdomain: readmeProjectData.subdomain as string,
+        stableVersion: readmeVersionData?.find(version => version.is_stable)?.version || '1.0',
+        readmeAPIKey,
+        disableMetrics: options.disableMetrics,
+        disableWebhook: options.disableWebhook,
+      });
+      return res.send(setupHtml);
+    } else if (req.path === '/webhook-test' && options.development) {
+      const email = req.query.email as string;
+      try {
+        const webhookData = await testVerifyWebhook(baseUrl, email, readmeProjectData.jwtSecret as string);
+        return res.json({ ...webhookData });
+      } catch (e) {
+        return res.status(400).json({ error: (e as Error).message });
+      }
+    }
+
+    try {
+      const user = await userFunction(req, getUser);
+      if (!user || !Object.keys(user).length || options.disableMetrics) return next();
+
+      const groupId = getGroupIdByApiKey(user, requestAPIKey);
+      if (!groupId) {
+        console.error(
+          usingManualAPIKey
+            ? 'The API key you passed in via `manualAPIKey` could not be found in the user object you provided.'
+            : 'Could not automatically find an API key in the request. You should pass the API key via `manualAPIKey` in the `getUser` function. Learn more here: https://docs.readme.com/main/docs/unified-snippet-docs#/getuserbyapikey',
+        );
+        return next();
+      }
+
+      log(
+        readmeAPIKey,
+        req,
+        res,
+        {
+          apiKey: groupId,
+          label: user.label ? user.label : user.name,
+          email: user.email,
+        },
+        options,
+      );
+    } catch (e) {
+      return next();
+    }
+    return next();
+  };
+};
+
+/**
+ * This method configures the API Key for your ReadMe project and returns middleware that will
+ * send supplied API requests to ReadMe Metrics and configure the ReadMe webhook.
+ *
+ * @see {@link https://readme.com/metrics}
+ * @see {@link https://docs.readme.com/main/docs/unified-snippet-docs}
+ * @param readmeApiKey The API key for your ReadMe project. This ensures your requests end up in
+ *    your dashboard. You can read more about the API key in
+ *    [our docs](https://docs.readme.com/reference/authentication).
+ */
+function ReadMe(key: string) {
```

Comment:
i'm curious about others' opinions on this but my understanding of JS conventions is that variable names should only be capitalized in the case of classes/constructors (e.g., `new Class()`, etc.) — thoughts on lowercasing it? what about calling this `metrics` or something like that?

Reply:
Sentry was my inspo here and they don't do that! https://docs.sentry.io/platforms/node/#configure

Reply:
interesting! sentry's SDK always gave class-like energy with the whole `Sentry.init` constructor pattern and it's apparently using [`AsyncLocalStorage`](https://github.com/getsentry/sentry-javascript/blob/1f1ccf13b69c683cea00cb2681320201727966e6/packages/node/src/async/hooks.ts#L20) to do that (which i don't totally understand...)

another idea here — what if you made this a class with a constructor? your example would look mostly the same (arguably a little cleaner + more intuitive to someone that expects JS conventions, in my opinion) and that way you can store [these values](https://github.com/readmeio/metrics-sdks/blob/f2fb3b61c3387225def145d5df3c494fd121f0d8/packages/node/src/lib/ReadMe.ts#L59-L61) as properties rather than globally which feels a little safer:

```diff
import { ReadMe } from 'readmeio';
- const readme = ReadMe(key);
+ const readme = new ReadMe(key);
app.use(readme.express(() => {}));
```

Reply:
agreed that it's a little weird to have a function be cased this way when that's generally how classes are defined + used.

what other popular sdks have exports like this?

Reply:
@kanadgupta I like the idea! It's definitely best practice from an initialization perspective, and would properly set expectations for consuming developers.

Reply:
I think if it's fine for Sentry it's understandable enough for our use case. I'm gonna move forward with it as it is! This PR is already way too long lived and we need to call off the rewriting somewhere. If someone else wants to take over this work feel free!